storage/src/main/java/org/apache/kafka/storage/internals/log/LogConfig.java

```diff
@@ -265,7 +266,12 @@ public Optional<String> serverConfigName(String configName) {
             .define(TopicConfig.LOCAL_LOG_RETENTION_MS_CONFIG, LONG, DEFAULT_LOCAL_RETENTION_MS, atLeast(-2), MEDIUM,
                 TopicConfig.LOCAL_LOG_RETENTION_MS_DOC)
             .define(TopicConfig.LOCAL_LOG_RETENTION_BYTES_CONFIG, LONG, DEFAULT_LOCAL_RETENTION_BYTES, atLeast(-2), MEDIUM,
-                TopicConfig.LOCAL_LOG_RETENTION_BYTES_DOC);
+                TopicConfig.LOCAL_LOG_RETENTION_BYTES_DOC)
+            // RemoteLogManagerConfig.REMOTE_LOG_STORAGE_SYSTEM_ENABLE_PROP is defined here to ensure that when system
+            // level remote storage functionality is disabled, topics cannot be configured to use remote storage.
+            .defineInternal(RemoteLogManagerConfig.REMOTE_LOG_STORAGE_SYSTEM_ENABLE_PROP, BOOLEAN,
+                    RemoteLogManagerConfig.DEFAULT_REMOTE_LOG_STORAGE_SYSTEM_ENABLE, null, MEDIUM,
+                    RemoteLogManagerConfig.REMOTE_LOG_STORAGE_SYSTEM_ENABLE_DOC);
```

Comment:
This still makes me feel it's not the correct place to put, while other configs are topic related configs. Could we pass in the `KafkaConfig` object into `LogConfig.validate`, i.e.
```
# currently
LogConfig.validate(config, kafkaConfig.map(_.extractLogConfigMap).getOrElse(Collections.emptyMap()))
# my suggestion
LogConfig.validate(config, kafkaConfig)
```
so that we don't need to make this `LogConfig` complicated? Thoughts @kamalcph @divijvaidya ?

Reply:
`KafkaConfig` class is in core module. `storage` module doesn't have dependency on the `core`. If we have to do the below change, then we have to make `core` as dependency for `storage`. Let me know if it is OK to add it:

```
public static LogConfig validateAndExtractLogConfig(kafka.server.KafkaConfig kafkaConfig) {
    Properties props = kafkaConfig.extractLogConfigMap();
    validateValues(props);
    if (kafkaConfig.isRemoteLogStorageSystemEnabled()) {
        validateRemoteStorageRetentionSize(props);
        validateRemoteStorageRetentionTime(props);
    }
    return new LogConfig(props);
}
```

Reply:
I remembered there will be cycle dependency issue there. So, if we only care about `REMOTE_LOG_STORAGE_SYSTEM_ENABLE_PROP` value currently, could we just pass that into `LogConfig.validate`? I.e. `LogConfig.validate(config, sysRemoteStorageEnabled)`

Reply:
I disagree that we just pass one specific config instead of generic kafkaConfig object, but in the interest of wrapping up this PR since it has undergone multiple revision, I am ok with this approach. We can always come back and refactor later.